src/rapini-mutation.ts

```diff
@@ -309,7 +309,7 @@ export function makeRapiniMutation() {
                             ts.factory.createIdentifier("context"),
                             ts.factory.createToken(ts.SyntaxKind.QuestionToken),
                             ts.factory.createTypeReferenceNode(
-                              ts.factory.createIdentifier("TContext"),
+                              ts.factory.createIdentifier("TContext | undefined"),
```

Comment:
The question token on line 310 covers the undefined type, this shouldn't be needed

Reply:
So in v4 it was giving an error saying context doesn't match type undefined it made the error go away. Coz when you generate it the type isn't there. As Context | undefined.

Reply:
ah okay so it's a v4 thing. Either way, We shouldn't do it like this. Should create a proper ast union type node like:
```ts
ts.factory.createUnionTypeNode([
  ts.factory.createTypeReferenceNode(ts.factory.createIdentifier("TContext")),
  ts.factory.createKeywordTypeNode(ts.SyntaxKind.UndefinedKeyword)
])
```

Reply:
Ooo i used a hacky solution classic beginner mistake. Okay thanks for your time. I will close the pr.